client/src/components/DataFiles/DataFilesProjectMembers/_cells/SystemRoleSelector.jsx

```diff
@@ -0,0 +1,100 @@
+import React, { useState, useEffect, createContext, useContext } from 'react';
+import { useQuery, useMutation } from 'react-query';
+import { useSelector } from 'react-redux';
+import Cookies from 'js-cookie';
+import fetch from 'cross-fetch';
+import DropdownSelector from '_common/DropdownSelector';
+import { Button } from 'reactstrap';
+import styles from '../DataFilesProjectMembers.module.scss';
+import LoadingSpinner from '_common/LoadingSpinner';
+
+const getSystemRole = async (projectId, username) => {
+  const url = `/api/projects/${projectId}/system-role/${username}/`;
+  const request = await fetch(url, {
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') },
+    credentials: 'same-origin',
+  });
+  const data = await request.json();
+  return data;
+};
+
+const setSystemRole = async (projectId, username, role) => {
+  const url = `/api/projects/${projectId}/members/`;
+  const request = await fetch(url, {
+    headers: { 'X-CSRFToken': Cookies.get('csrftoken') },
+    credentials: 'same-origin',
+    method: 'PATCH',
+    body: JSON.stringify({
+      action: 'change_system_role',
+      username,
+      newRole: role,
+    }),
+  });
+  const data = await request.json();
+  return data;
+};
+
+export const useSystemRole = (projectId, username) => {
+  const query = useQuery(['system-role', projectId, username], () =>
+    getSystemRole(projectId, username)
+  );
+  const mutation = useMutation(async (role) => {
+    await setSystemRole(projectId, username, role);
+    query.refetch();
+  });
+  return { query, mutation };
+};
+
+const SystemRoleSelector = ({ projectId, username }) => {
+  const authenticatedUser = useSelector(
+    (state) => state.authenticatedUser.user.username
+  );
+  const { query: authenticatedUserQuery } = useSystemRole(
+    projectId,
+    authenticatedUser
+  );
+  const currentUserRole = authenticatedUserQuery.data?.role;
+
+  const {
+    query: { data, isLoading, isFetching, error },
+    mutation: { mutate: setSystemRole, isLoading: isMutating },
+  } = useSystemRole(projectId, username);
+  const [selectedRole, setSelectedRole] = useState(data?.role);
+  useEffect(() => setSelectedRole(data?.role), [data?.role]);
+
+  if (isLoading || authenticatedUserQuery.isLoading)
+    return <LoadingSpinner placement="inline" />;
+  if (error) return <span>Error</span>;
+  //Only owners/admins can change roles;
+  // owner roles cannot be changed except using the Transfer mechanism;
+  // users cannot change their own roles.
+  if (
+    data.role === 'OWNER' ||
+    username === authenticatedUser ||
+    !['OWNER', 'ADMIN'].includes(currentUserRole)
+  )
+    return <span>{data.role}</span>;
+  return (
+    <div style={{ display: 'inline-flex' }}>
+      <DropdownSelector
```

Comment:
We should disable the dropdown while `Transfer Ownership` state is active
<img width="479" alt="Screen Shot 2022-05-11 at 2 11 33 PM" src="https://user-images.githubusercontent.com/20326896/167927813-1acd397b-cff2-49f1-a6dd-93d509d82af4.png">


Also, "Transfer Ownership" overflows around ~987px

Reply:
This overflow can be made less ugly with [three lines of CSS either copied in or via a mixin in a `.scss` file](https://github.com/TACC/Core-Portal/blob/main/client/src/styles/tools/mixins.scss). If after this the use only sees "Transf..." let's just make the button wide enough to show "Transfer...".

Reply:
Is there a compelling reason to disable the dropdown while transferring ownership? It's going to introduce some really ugly prop-drilling spaghetti, and I don't see a way for it to actually break anything since hitting the "transfer" confirmation will reset permissions anyway.

Reply:
This is an existing bug (you can see it for yourself in any prod portal). I think we should make a new ticket for it since I tried a few obvious-seeming fixes and they all failed in weird ways. FWIW if you shrink the window past ~900px it cuts off the "manage team members" link, so this bug will only affect users who open the modal, hit "change ownership", and then mess with the viewport.

Reply:
For the button overflow: new ticket, new PR https://github.com/TACC/Core-Portal/pull/641.